packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1100,17 +1123,15 @@ export class GarbageCollector implements IGarbageCollector {
         }
 
         const gcSnapshotTree = snapshot.trees[gcTreeKey];
-        if (gcSnapshotTree !== undefined && this.trackGCState) {
+        if (gcSnapshotTree !== undefined) {
             const latestGCData = await getGCDataFromSnapshot(
                 gcSnapshotTree,
                 readAndParseBlob,
             );
-            this.latestSummaryData = {
-                serializedGCState: JSON.stringify(generateSortedGCState(latestGCData.gcState)),
-                serializedTombstones: JSON.stringify(latestGCData.tombstones),
-            };
+            this.updateStateFromSnapshotData(latestGCData);
```

Comment:
Note: For future posterity, this is the key defining fix that we care about.

Reply:
I will add a comment.